### PR TITLE
Update template with tag pages

### PR DESCRIPTION
**Pull request by frankieroberto**
This adds tag pages to the template, following [the instructions from the eleventy plugin](https://x-govuk.github.io/govuk-eleventy-plugin/tagging/).

### CODE REVIEW

**Comment by frankieroberto** [comment]
@paulrobertlloyd this works, but we still have the issue with all posts being tagged `post`.

Screenshot 2023-09-18 at 10 18 07

 I tried fixing this by removing the `tags: ["post"]` from `posts.json`, but then that means that the homepage listing no longer works, as that's iterating through `collections.post`. I could change that to `collections.all`, which I think makes sense to do, but then the tag pages themselves appear in the list. I tried to fix that by removing those pages from the `all` collection using `eleventyExcludeFromCollections: true` but then this also empties the `collections.tagPages` collection as that collection [is built by iterating over all pages and filtering for ones using the tag layout](https://github.com/x-govuk/govuk-eleventy-plugin/blob/main/lib/collections/tag-pages.js))...
 
The way we resolved this on the BAT design history was to build the `tagPages` collection (called `allTags` there) by iterating over all pages and then [building an array of unique tags from the post metadata](https://github.com/DFE-Digital/bat-design-history/blob/main/.eleventy.js#L109-L143). Is it worth considering switching to that approach in the eleventy-plugin, or is there a better way? 🤔

**Comment by paulrobertlloyd** [comment]
> Is it worth considering switching to that approach in the eleventy-plugin, or is there a better way?

If you mean using the collection method you’ve used on the BAT Design History as the means of creating the tags collection in the Eleventy plugin, seems to make sense to me. Let’s give it a shot!

**Comment by frankieroberto** [comment]
@paulrobertlloyd got this working! I realised that another downside of not adding the tag pages to `collections.all` is that using `eleventyNavigation` for the breadcrumb on tag pages no longer works... 🤦  (the way that plugin works is super confusing!).  So I've hardcoded the breadcrumbs on those templates instead - which is what I also ended up doing on the BAT design history.

**Comment by paulrobertlloyd** [comment]
> the way that plugin works is super confusing

I agree. Almost think it needs to be completely rearchitected and rethought.

> I've hardcoded the breadcrumbs on those templates instead

I wonder if this can be somewhat automated. A site may have a `parentSite` and I think the ‘Home’ text is defined in `homeKey`. Not sure if `options` is available globally but maybe something like this:

```njk
{% block beforeContent %}
 {{ govukBreadcrumbs({
   items: [{
     text: options.parentSite.name,
     href: options.parentSite.url
   } if options.parentSite, {
     text: options.homeKey,
     href: "/"
   },
   {
     text: "Tags",
     href: "/tags"
   }]
 }) }}{% endblock %}
```

**Comment by frankieroberto** [comment]
@paulrobertlloyd good idea! Done in b3a306a6d3dbecb60923ce1f0f56f920481a226c and tested it locally with a parentSite set. Works perfectly, thanks!

File: app/_layouts/tag.njk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{% extends "layouts/tag.njk" %}
+
+{% block beforeContent %}
+  {{ govukBreadcrumbs({
+    items: [
+      {
+        text: options.parentSite.name,
+        href: options.parentSite.url
+       } if options.parentSite,
```
Comment (paulrobertlloyd) [inline app/_layouts/tag.njk]:
```suggestion
      } if options.parentSite,
```

File: app/_layouts/tags.njk [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% extends "layouts/tags.njk" %}
+{% block beforeContent %}
+  {{ govukBreadcrumbs({
+    items: [
+      {
+        text: options.parentSite.name,
+        href: options.parentSite.url
+       } if options.parentSite,
```
Comment (paulrobertlloyd) [inline app/_layouts/tags.njk]:
```suggestion
      } if options.parentSite,
```

**Comment by frankieroberto** [comment]
@paulrobertlloyd happy if I merge this now?  There's a version number in the `package.json` we could bump, but as this isn't an npm module, but just a GitHub template repo, there might be less need to bump the version number?

**Comment by paulrobertlloyd** [comment]
Yeah, squash and merge I think.

There isn’t really any version number for this project, as it isn’t a package as you say. I do wonder if we should have some sort of release tracking though, even if that is just tagging commits and writing release notes?

**Pull request merged**